onnx/defs/tensor/defs.cc

```diff
@@ -252,14 +243,12 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
-          if (ctx.getInputType(0)->tensor_type().has_shape()) {
-            ctx.getOutputType(0)
-                ->mutable_tensor_type()
-                ->mutable_shape()
-                ->add_dim()
-                ->set_dim_value(
-                    ctx.getInputType(0)->tensor_type().shape().dim_size());
-          }
+          ctx.getOutputType(0)
+              ->mutable_tensor_type()
+              ->mutable_shape()
+              ->add_dim()
+              ->set_dim_value(
+                  ctx.getInputType(0)->tensor_type().shape().dim_size());
```

Comment:
what will be set if "ctx.getInputType(0)->tensor_type().has_shape() is false"? "0"? if that's the case it's wrong. you may always add a dim, but the value should not be set in that case (ctx.getInputType(0)->tensor_type().has_shape() is false).

Reply:
Well I just removed the duplicate check - in the line previous to this - the if condition invokes `hasNInputShapes()` which invokes `hasInputShape()` which basically checks if `has_shape()` is true along with a bunch of other stuff. So this if check is redundant and so I removed it.

One thing that I can improve is actually to add the dimension even before the first if check (rank inference) because output is always 1D. I can fill the value with input rank if shape is known.